.travis.yml

```diff
@@ -1,8 +1,35 @@
+---
 language: python
 python:
-  - "3.7"
+  - '3.7'
 install:
   - pip install -r requirements.txt -r requirements-dev.txt
 script:
   - make lint
   - make unittest
+
+before_deploy:
+  - >
+    if [[ -n "$TRAVIS_TAG" ]] && [[ "${TRAVIS_TAG}" != "$(<VERSION)" ]]; then
+        >&2 echo "ERROR: You have marked a release by applying a Git tag and your VERSION file doesn't match it"
+        >&2 echo "Tip: Don't tag if you want to deploy only to test.pypi.org"
+        false
+    fi
+  - export VERSION=${TRAVIS_TAG:-$(<VERSION).dev${TRAVIS_BUILD_NUMBER}}
+
+deploy:
+  - provider: pypi
+    user: __token__
+    password: "$TEST_PYPI_ORG_TOKEN"
+    distributions: sdist bdist_wheel
+    server: https://test.pypi.org/legacy/
+    "on":
+      all_branches: true
+      tags: false
+
+env:
+  global:
+    # TEST_PYPI_ORG_TOKEN
+    - secure: TUQef9pZbFtOBGLQkAT0pzRmYjTR6LJOucM2x+kMfGfueN1khiwx/LmcShgCha+7kH43v0/1q9Iyk6+xQIfe7TfkL5vxD90s3Ej+ioeSK0BeqaO6Tf/lnkU2AaqpJolJC+C35/whQ4d051vhUqefPN3zF0MSXrf8rYgHt59rDIwzumbJQ3+x3g+8UOwWqdhIwrt3lVWdnYaDzkIGVIys6NoUpeDLl2RvWRKT7/8SQAJ2GKS1KffzMlnk5hAP5O2RzgeB5o9n1I/hZKoS44PIPZwRzmkbl0r8wE7XRhvwgOcKxNY1lqsTpVMR2CQrNBvqeU/mBpREjsWV33s+kctl9qd/DbHVpFRGMnGbRPe8zgUNF+spr7DU5p4343KMuuuE2V9m8VGM0vI3z9ZPgVqI6dRju3M2hFMxpX2k4iVQK+mv9rvx0WkvYUEvrFE3v0D1IQ3BgOmC+cykdxeTRZPZfptNXSVQWDfO9ihKUakUyICZzO/4HrxhfUqQGcFhNmwbVj2VQW/9vFhew8CCLzfAUIzLYb+iyEsO57CHf1/5a4u3ThANUUI9LCEHQIXzhLSqdF+H3ES4LwdzTNeHxcFMShEyPs0iD71tYUXX14W6/fpI1fJukbaa1C8Af/4OKNcXS6s/88PIVQ38Jda+tp6G4BQyQAR1raTZGUMmySuvM9M=
```

Comment:
is this the token? is it okay to put it here?

Reply:
It's a token, but encrypted by Travis. And it only gets properly decrypted within the context of original repository, so forks won't get it.

So yes, it's reasonably safe.

https://docs.travis-ci.com/user/deployment/pypi/